AppController/djinn.rb

```diff
@@ -1949,14 +1949,35 @@ def update(apps, secret)
     }
     failed_apps.each { |app|
       apps_to_restart.delete(app)
-      stop_app(app, @@secret)
-      Djinn.log_error("Disabled #{app} since language doesn't match our record.")
+      Djinn.log_error("Ignoring update to #{app} since language doesn't match our records.")
     }
 
     # Make sure we have the latest code deployed.
     apps.each { |appid|
-      APPS_LOCK.synchronize {
-        setup_app_dir(appid, true)
+      next if failed_apps.include?(appid)
+
+      # Let's make sure the application is enabled.
+      uac = UserAppClient.new(my_node.private_ip, @@secret)
+      RETRIES.downto(0) { |tries|
+        begin
+          result = uac.enable_app(appid)
+          Djinn.log_debug("enable_app returned #{result}.")
+          APPS_LOCK.synchronize {
+            setup_app_dir(appid, true)
+          }
+          break
+        rescue FailedNodeException
```

Comment:
Please only rescue the code that can throw the exception. Otherwise, it's more difficult to track down the problem when we encounter the error.